ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,425 @@
+        }
+        drawLine(canvas, shape[i][0], shape[i][shape[0].size()-1], pose, resolution, origin_x, origin_y, color);
+    }
+
+    // paint entity center
+
+    int x_p_ent = origin_x + (int)(pose.t.x * resolution);
+    int y_p_ent = origin_y - (int)(pose.t.y * resolution);
+
+    cv::Point Entity_center(x_p_ent, y_p_ent);
+    cv::circle(canvas, Entity_center, 3, color, cv::FILLED);
+
+}
+// ----------------------------------------------------------------------------------------------------
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "ed_fitter");
+    ros::NodeHandle nh;
+
+    if (argc != 4)
+    {
+        usage();
+        return 1;
+    }
+
+    std::string model_name = argv[2];
+
+    ed::WorldModel world_model;
+    if (!loadWorldModel(model_name, world_model))
+        return 1;
+
+    tue::filesystem::Path path = argv[1];
+    if (!path.exists())
+    {
+        std::cerr << "Path '" << path << "' does not exist." << std::endl;
+        return 1;
+    }
+
+    std::string entity_id = argv[3];
+
+
+    tue::filesystem::Crawler crawler;
+
+    if (path.isDirectory())
+        crawler.setRootPath(path);
+    else
+        crawler.setRootPath(path.parentPath());
+
+    Updater updater;
+    Fitter fitter;
+
+    std::vector<Snapshot> snapshots;
+    unsigned int i_snapshot = 0;
+
+    while(ros::ok())
+    {
+        if (i_snapshot >= snapshots.size())
+        {
+            bool file_found = true;
+            tue::filesystem::Path filename;
+
+            if (path.isRegularFile() && snapshots.empty())
+                filename = path;
+            else
+            {
+                file_found = false;
+                while (crawler.nextPath(filename))
+                {
+                    if (filename.extension() == ".json")
+                    {
+                        file_found = true;
+                        break;
+                    }
+                }
+            }
+
+            if (file_found)
+            {
+                i_snapshot = snapshots.size();
+                snapshots.push_back(Snapshot());
+                Snapshot& snapshot = snapshots.back();
+                snapshot.world_model = world_model;
+
+                if (!readImage(filename.string(), snapshot.image, snapshot.sensor_pose,
+                               snapshot.world_model, snapshot.area_description))
+                {
+                    std::cerr << "Could not read " << filename << std::endl;
+                    snapshots.pop_back();
+                    continue;
+                }
+                else
+                {
+//                    std::cout << "Successfully loaded " << filename << std::endl;
+                }
+            }
+            else
+            {
+                if (snapshots.empty())
+                    break;
+
+                i_snapshot = snapshots.size() - 1;
+            }
+        }
```

Comment:
A lot of indentation, could this be simplified?

Reply:
probably not